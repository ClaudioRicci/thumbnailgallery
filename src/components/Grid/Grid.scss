body {
  font-family: "Varela", sans-serif;
  background: #ccc;
}

h2 {
  font-size: 1.2em;
}

h3 {
  font-size: 1.1em;
  line-height: 0;
}

p {
  font-size: 1em;
}

h2,
p {
  line-height: 0.6em;
}

.centered {
  margin: 0 auto;
  padding: 0 1em;
}

.cards {
  display: flex;
  flex-wrap: wrap;
}

@media screen and (min-width: 40em) {
  .cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }
}

@media screen and (min-width: 52em) {
  .centered {
    max-width: 70em;
  }
}
