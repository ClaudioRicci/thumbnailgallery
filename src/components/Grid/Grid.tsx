import React, { useState, useEffect, memo } from "react";
import Card from "../Card/Card";
import Header from "../Header/Header";
import LoadingCircle from "../LoadingCircle/LoadingCircle";
import axios from "axios";
import "./Grid.scss";

function Grid() {
  const [data, setData] = useState({ images: [] });
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false);
  const url = "http://www.splashbase.co/api/v1/images/search?query=cars";
  useEffect(() => {
    const fetchData = async () => {
      setIsError(false);
      setIsLoading(true);
      try {
        const result = await axios(url);
        setData(result.data);
      } catch (error) {
        setIsError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [url]);
  return (
    <main className="grid" data-testid="grid">
      {isError && <LoadingCircle message="Whoops! An error occured" />}
      {isLoading ? (
        <LoadingCircle message="Loading..." />
      ) : (
        <>
          {!isError && (
            <section className="grid__cards">
              <Header title="Thumbnail Gallery" />
              {data.images.map(item => (
                <Card
                  key={item.id}
                  thumbnail_url={item.url}
                  large_url={item.large_url}
                  copyright={item.copyright}
                  site={item.site}
                />
              ))}
            </section>
          )}
        </>
      )}
    </main>
  );
}

export default memo(Grid);
