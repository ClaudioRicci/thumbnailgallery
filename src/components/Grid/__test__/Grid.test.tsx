import React from "react";
import ReactDOM from "react-dom";
import Grid from "../Grid";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Grid></Grid>, div);
});

test("Renders Grid Correctly", done => {
  const { getByTestId } = render(<Grid />);
  expect(getByTestId("grid")).toBeInTheDocument();
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Grid></Grid>).toJSON();
  expect(tree).toMatchSnapshot();
});
