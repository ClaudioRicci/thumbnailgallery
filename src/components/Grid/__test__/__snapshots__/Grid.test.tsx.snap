// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<main
  className="grid"
  data-testid="grid"
>
  <section
    className="grid__cards"
  >
    <header
      data-testid="header"
    >
      <svg
        className="logo"
        height="48"
        viewBox="0 0 160 48"
        width="160"
      >
        <path
          d="M154.7 18a2.65 2.65 0 1 0 5.3 0 2.65 2.65 0 0 0-5.3 0zm-8.4-2.4h-5.6l7.6 13.2-7.6 13.2h5.6l7.6-13.2-7.6-13.2zm-10.8 12c0 1.6-1.3 2.6-3.3 2.6-1.7 0-2.8-.8-2.8-2 0-1.1.8-1.8 2.6-2.2l3.5-.4v2zm4.7.1v-5.9c0-2.8-.4-3.7-1.9-4.8a8.4 8.4 0 0 0-5.4-1.6c-2.7 0-5 .6-7.3 2l2 3.6c1.9-1 3.7-1.6 5.2-1.6 1.2 0 2 .5 2.4 1.2.2.4.2.3.2 1.5-4 .3-6.8.8-8.4 1.8a4.9 4.9 0 0 0-2.3 4.2c0 3.4 3 6 7 6 2.1 0 3.5-.6 4.5-2 1 1.3 2.1 2 3.5 2h.4l2-3.5c-1.6-.4-1.9-.8-1.9-2.9zm-78 1.3l-1.9.3c-1.8 0-2.3-.6-2.3-2.9v-6.9h4.2v-3.9H58V10h-4.8v5.6h-2.8v4h2.8v8.9c0 2 .1 2.8.7 3.5.8 1 2.3 1.6 4.5 1.6 1.4 0 2.7-.1 3.8-.5V29zm6-4.2c0 3 2.1 5 4.9 5 2.7 0 4.8-2 4.8-5s-2.1-5.1-4.9-5.1c-2.8 0-4.8 2-4.8 5zm14.3 0c0 5.3-4 9.3-9.4 9.3-5.5 0-9.5-4-9.5-9.3 0-5.4 4-9.4 9.5-9.4 5.4 0 9.4 4 9.4 9.4zm-47.2 0c0 3 2 5 4.8 5 2.7 0 4.8-2 4.8-5s-2-5.1-4.8-5.1-4.8 2-4.8 5zm14.3 0c0 5.3-4 9.3-9.5 9.3s-9.5-4-9.5-9.3c0-5.4 4-9.4 9.5-9.4 5.4 0 9.5 4 9.5 9.4zm69.6 8.8h-5.6l-2.9-10.8-2.6 10.8h-5.8l-5-18h5l3 12.2 2.7-12.2h5.2l3 12.2 3-12.2h4.9l-5 18zm-34.4 0v-18h4.5l.3 2c1.5-1.5 2.7-2.2 4.2-2.2.5 0 1 0 1.8.2v4.5h-.7c-1.7 0-2.8.3-3.8 1.2-1.2 1-1.5 1.8-1.5 4.3v8h-4.8zm-61.6 0v-9.7c0-2.9-1-4.2-3.2-4.2-1.2 0-2.3.6-3 1.6-.5.8-.7 1.7-.7 3.3v9h-4.7v-9.2c0-1.6-.2-2.6-.6-3.3a3.1 3.1 0 0 0-2.7-1.3c-2.3 0-3.6 1.5-3.6 4.4v9.4H0v-18h4.4l.3 1.5c1.8-1.3 3-1.7 4.7-1.7 2.3 0 4 .8 5.7 2.7 2-2 3.8-2.7 6-2.7 2.4 0 4.4 1 5.6 2.7 1 1.3 1.2 2.4 1.2 5v10.5h-4.7z"
        />
      </svg>
      <h1>
        Thumbnail Gallery
      </h1>
      <span>
        Click on 
        <svg
          data-testid="searchIcon"
          height="36"
          viewBox="0 0 52.966 52.966"
          width="120"
        >
          <path
            d="M28.983,20h-6v-6c0-0.552-0.448-1-1-1s-1,0.448-1,1v6h-6c-0.552,0-1,0.448-1,1s0.448,1,1,1h6v6c0,0.552,0.448,1,1,1 s1-0.448,1-1v-6h6c0.552,0,1-0.448,1-1S29.535,20,28.983,20z"
          />
          <path
            d="M51.704,51.273L36.845,35.82c3.79-3.801,6.138-9.041,6.138-14.82c0-11.58-9.42-21-21-21s-21,9.42-21,21s9.42,21,21,21 c5.083,0,9.748-1.817,13.384-4.832l14.895,15.491c0.196,0.205,0.458,0.307,0.721,0.307c0.25,0,0.499-0.093,0.693-0.279 C52.074,52.304,52.086,51.671,51.704,51.273z M2.983,21c0-10.477,8.523-19,19-19s19,8.523,19,19s-8.523,19-19,19 S2.983,31.477,2.983,21z"
          />
        </svg>
         to enlarge
      </span>
    </header>
  </section>
</main>
`;
